code/game/objects/items/devices/scanners.dm

```diff
@@ -632,3 +632,253 @@ REAGENT SCANNER
+	playsound(src, 'sound/machines/defib_saftyOn.ogg', 50, 0)
+	update_icon()
+
+/obj/item/bodyanalyzer/update_icon(printing = FALSE)
+	overlays.Cut()
+	var/percent = power_supply.percent()
+	if(ready)
+		icon_state = "bodyanalyzer_1"
+	else
+		icon_state = "bodyanalyzer_2"
+	
+	var/overlayid = round(percent / 10)
+	overlayid = "bodyanalyzer_charge[overlayid]"
+	overlays += icon(icon, overlayid)
+
+	if(printing)
+		overlays += icon(icon, "bodyanalyzer_printing")
+
+/obj/item/bodyanalyzer/attack(mob/living/M, mob/living/carbon/human/user)
+	if(user.incapacitated() || !user.Adjacent(M))
+		return
+	
+	if(!ready)
+		to_chat(user, "<span class='notice'>The scanner beeps angrily at you! It's currently recharging - [round((time_to_use - world.time) * 0.1)] seconds remaining.")
+		playsound(user.loc, 'sound/machines/buzz-sigh.ogg', 50, 1)
+		return
+	
+	if(power_supply.charge >= usecharge)
+		mobScan(M, user)
+	else
+		to_chat(user, "<span class='notice'>The scanner beeps angrily at you! It's out of charge!")
+		playsound(user.loc, 'sound/machines/buzz-sigh.ogg', 50, 1)
+	
+/obj/item/bodyanalyzer/proc/mobScan(mob/living/M, mob/user)
+	if(ishuman(M))
+		var/report = generate_printing_text(M, user)
+		user.visible_message("[user] begins scanning [M] with [src].", "You begin scanning [M].")
+		if(do_after(user, 100, target = M))
+			var/obj/item/paper/printout = new
+			printout.info = report
+			printout.name = "Scan report - [M.name]"
+			playsound(user.loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+			user.put_in_hands(printout)
+			time_to_use = world.time + 600
+			power_supply.use(usecharge)
+			ready = FALSE
+			update_icon(TRUE)
+			addtimer(CALLBACK(src, /obj/item/bodyanalyzer/.proc/setReady), 600)
+			addtimer(CALLBACK(src, /obj/item/bodyanalyzer/.proc/update_icon), 20)
+
+	else if(iscorgi(M) && M.stat == DEAD)
+		to_chat(user, "<span class='notice'>You wonder if [M.p_they()] was a good dog. <b>[src] tells you they were the best...</b></span>") // :'(
+		playsound(loc, 'sound/machines/ping.ogg', 50, 0)
+		ready = FALSE
+		addtimer(CALLBACK(src, /obj/item/bodyanalyzer/.proc/setReady), 600)
+		time_to_use = world.time + 600
+	else
+		to_chat(user, "<span class='notice'>Scanning error detected. Invalid specimen.</span>")
+		
+//Unashamedly ripped from adv_med.dm
+/obj/item/bodyanalyzer/proc/generate_printing_text(mob/living/M, mob/user)
+	var/dat = ""
+	var/mob/living/carbon/human/target = M
+	
+	dat = "<font color='blue'><b>Target Statistics:</b></font><br>"
+	var/t1
+	switch(target.stat) // obvious, see what their status is
+		if(CONSCIOUS)
+			t1 = "Conscious"
+		if(UNCONSCIOUS)
+			t1 = "Unconscious"
+		else
+			t1 = "*dead*"
+	dat += "[target.health > 50 ? "<font color='blue'>" : "<font color='red'>"]\tHealth %: [target.health], ([t1])</font><br>"
+
+	var/found_disease = FALSE
+	for(var/thing in target.viruses)
+		var/datum/disease/D = thing
+		if(D.visibility_flags & HIDDEN_SCANNER || D.visibility_flags & HIDDEN_PANDEMIC)

```

Comment:
Tiny thing, but maybe rework so you're not just checking the same var twice...

Reply:
I mean, it's checking that two different bits are on. But I could just do `if(D.visibility_flags) as there's only two to worry about.